Assets/MRTK/Providers/LeapMotion/LeapMotionArticulatedHand.cs

```diff
@@ -0,0 +1,315 @@
+        }
+
+        /// <summary>
+        /// Converts a TrackedHandJoint to a Leap AttachmentPointFlag. An AttachmentPointFlag is Leap's version of MRTK's TrackedHandJoint.
+        /// </summary>
+        /// <param name="joint">TrackedHandJoint to be mapped to a Leap AttachmentPointFlag</param>
+        /// <returns>Leap Motion AttachmentPointFlag pose</returns>
+        private AttachmentPointFlags ConvertMRTKJointToLeapJoint(TrackedHandJoint joint)
+        {
+            switch (joint)
+            {
+                case TrackedHandJoint.Palm: return AttachmentPointFlags.Palm;
+                case TrackedHandJoint.Wrist: return AttachmentPointFlags.Wrist;
+
+                case TrackedHandJoint.ThumbProximalJoint: return AttachmentPointFlags.ThumbProximalJoint;
+                case TrackedHandJoint.ThumbDistalJoint: return AttachmentPointFlags.ThumbDistalJoint;
+                case TrackedHandJoint.ThumbTip: return AttachmentPointFlags.ThumbTip;
+
+                case TrackedHandJoint.IndexKnuckle: return AttachmentPointFlags.IndexKnuckle;
+                case TrackedHandJoint.IndexMiddleJoint: return AttachmentPointFlags.IndexMiddleJoint;
+                case TrackedHandJoint.IndexDistalJoint: return AttachmentPointFlags.IndexDistalJoint;
+                case TrackedHandJoint.IndexTip: return AttachmentPointFlags.IndexTip;
+
+                case TrackedHandJoint.MiddleKnuckle: return AttachmentPointFlags.MiddleKnuckle;
+                case TrackedHandJoint.MiddleMiddleJoint: return AttachmentPointFlags.MiddleMiddleJoint;
+                case TrackedHandJoint.MiddleDistalJoint: return AttachmentPointFlags.MiddleDistalJoint;
+                case TrackedHandJoint.MiddleTip: return AttachmentPointFlags.MiddleTip;
+
+                case TrackedHandJoint.RingKnuckle: return AttachmentPointFlags.RingKnuckle;
+                case TrackedHandJoint.RingMiddleJoint: return AttachmentPointFlags.RingMiddleJoint;
+                case TrackedHandJoint.RingDistalJoint: return AttachmentPointFlags.RingDistalJoint;
+                case TrackedHandJoint.RingTip: return AttachmentPointFlags.RingTip;
+
+                case TrackedHandJoint.PinkyKnuckle: return AttachmentPointFlags.PinkyKnuckle;
+                case TrackedHandJoint.PinkyMiddleJoint: return AttachmentPointFlags.PinkyMiddleJoint;
+                case TrackedHandJoint.PinkyDistalJoint: return AttachmentPointFlags.PinkyDistalJoint;
+                case TrackedHandJoint.PinkyTip: return AttachmentPointFlags.PinkyTip;
+
+                // Metacarpals are not included in AttachmentPointFlags
+                default: return AttachmentPointFlags.Wrist;
+            }
+        }
+
+        public void UpdateState()
+        {
+            // Populate the jointPoses 
+            foreach (TrackedHandJoint handJoint in TrackedHandJointEnum)
+            {
+                if (TryGetJoint(handJoint, out MixedRealityPose pose))
```

Comment:
It feels like the contents of `TryGetJoint` should be moved to a different method which grabs and stores the jointPoses from the Leap APIs (which would be called here instead).

The reason I say this is because `TryGetJoint` is a public method. We have some scripts that ask `IMixedRealityHand`s for their joints, which means that each time it gets called we'll be running through the steps of grabbing the data from the `AttachmentHand`, converting it, etc, even though we've cached the result here.

For example, in the WMR articulated hand class, we simply return this cached value:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/614f4a069fd108caa31dc6eaa0d6bb1b739065c8/Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/WindowsMixedRealityArticulatedHand.cs#L63

Reply:
Moved the leap joint conversion to another method and incorporated the suggestion.